ow_sim_tests/scripts/test_camera_set_exposure.py

```diff
@@ -0,0 +1,121 @@
+    camera_set_exposure_result = test_action(self,
+      'CameraSetExposure', owl_msgs.msg.CameraSetExposureAction,
+      owl_msgs.msg.CameraSetExposureGoal(
+        automatic = True,
+        exposure = -1
+      ),
+      TEST_NAME, 'test_01_camera_set_exposure',
+      server_timeout = 50.0 # (seconds) first action call needs longer timeout
+    )
+
+  # Move the antenna so the camera can see the robot arm.
+  def test_02_pan_tilt_move_cartesian(self):
+    pan_tilt_move_cartesian_result = test_action(self,
+      'PanTiltMoveCartesian', owl_msgs.msg.PanTiltMoveCartesianAction,
+      owl_msgs.msg.PanTiltMoveCartesianGoal(
+        frame = 1,
+        point = Point(0, 0, 0)
+      ),
+      TEST_NAME, "test_02_pan_tilt_move_cartesian"
+    )
+  
+  def test_03_camera_capture(self):
+    camera_capture_result = test_action(self,
+      'CameraCapture', owl_msgs.msg.CameraCaptureAction,
+      owl_msgs.msg.CameraCaptureGoal(),
+      TEST_NAME, 'test_03_camera_capture'
+    )
+
+  def test_04_camera_set_exposure(self):
+    camera_set_exposure_result = test_action(self,
+      'CameraSetExposure', owl_msgs.msg.CameraSetExposureAction,
+      owl_msgs.msg.CameraSetExposureGoal(
+        automatic = False,
+        exposure = 0.01
+      ),
+      TEST_NAME, 'test_04_camera_set_exposure'
+    )
+
+  def test_05_camera_capture(self):
+    camera_capture_result = test_action(self,
+      'CameraCapture', owl_msgs.msg.CameraCaptureAction,
+      owl_msgs.msg.CameraCaptureGoal(),
+      TEST_NAME, 'test_05_camera_capture'
+    )
+
+  def test_06_camera_set_exposure(self):
+    camera_set_exposure_result = test_action(self,
+      'CameraSetExposure', owl_msgs.msg.CameraSetExposureAction,
+      owl_msgs.msg.CameraSetExposureGoal(
+        automatic = False,
+        exposure = 0.02
+      ),
+      TEST_NAME, 'test_06_camera_set_exposure'
+    )
+
+  def test_07_camera_capture(self):
+    camera_capture_result = test_action(self,
+      'CameraCapture', owl_msgs.msg.CameraCaptureAction,
+      owl_msgs.msg.CameraCaptureGoal(),
+      TEST_NAME, 'test_07_camera_capture'
+    )
+
+  def test_08_camera_set_exposure(self):
+    camera_set_exposure_result = test_action(self,
+      'CameraSetExposure', owl_msgs.msg.CameraSetExposureAction,
+      owl_msgs.msg.CameraSetExposureGoal(
+        automatic = False,
+        exposure = 0.04
+      ),
+      TEST_NAME, 'test_08_camera_set_exposure'
+    )
+
+  def test_09_camera_capture(self):
+    camera_capture_result = test_action(self,
+      'CameraCapture', owl_msgs.msg.CameraCaptureAction,
+      owl_msgs.msg.CameraCaptureGoal(),
+      TEST_NAME, 'test_09_camera_capture'
+    )
+
+  def test_10_camera_set_exposure(self):
+    camera_set_exposure_result = test_action(self,
+      'CameraSetExposure', owl_msgs.msg.CameraSetExposureAction,
+      owl_msgs.msg.CameraSetExposureGoal(
+        automatic = False,
+        exposure = 0.05
+      ),
+      TEST_NAME, 'test_10_camera_set_exposure'
+    )
+
```

Comment:
You set the exposure in test_10, why not capture a picture for test_11 to finish off the test?

Reply:
Oh, I forgot to add one more camera_capture at the end, thank you for pointing it out.